src/Agent.Worker/TestResults/TestRunPublisher.cs

```diff
@@ -268,7 +269,7 @@ private TestAttachmentRequestModel GetAttachmentRequestModel(string attachment)
             {
```

Comment:
As discussed if file.Exists is not needed remove it and always use "AttachmentExceededMaximum" if the size exceeded